### ISSUE TITLE
How to boot Rescuezilla as a rescue partition, rather from USB stick?

### ISSUE THREAD

**Issue by AE720** [rescuezilla/rescuezilla#207]
This may show my ignorance with how GRUB work but I would like a way to either simply install rescuezilla to a new partition on a disk that you're working on so that instead of booting from removable media, it will show up on the boot list. This might also compliment some features such as the ability to have preset settings as now you can make those settings and simply copy them over to a rewritable partition. I'm aware there might be risks especially to the end user and if Windows is involved but I know I would like that alternative. (I only have two easily accessible usb slots on my home server and I have to load into RAM by the kb so I can unplug it and plug the mouse back in afterwards)

As usual, I understand if it's not safe, feasible and it can be a very low priority.

**Comment by shasheene** [comment]
It's possible to to [configure GRUB2 to boot the Rescuezilla ISO image that resides on a partition of your hard drive](https://www.linuxbabe.com/desktop-linux/boot-from-iso-files-using-grub2-boot-loader). The instructions I've linked to claims to work for both legacy MBR boot and EFI boot, but only with Secure Boot disabled. Here's a [video](https://www.youtube.com/watch?v=Z35zyDpMIMM) describing the same instructions.

I successfully run the instructions on a virtual machine that happened to be using legacy MBR boot. Here's the `/etc/grub.d/40_custom` file with settings relevant for Rescuezilla:

```
#!/bin/sh
exec tail -n +3 $0
# This file provides an easy way to add custom menu entries.  Simply type the
# menu entries you want to add after this comment.  Be careful not to change
# the 'exec tail' line above.

menuentry "rescuezilla-2.1.3-64bit.groovy.iso" {
  insmod ext2
  set isofile="/home/USERNAME/Downloads/rescuezilla-2.1.3-64bit.groovy.iso"
  loopback loop (hd0,5)$isofile
  linux (loop)/casper/vmlinuz boot=casper iso-scan/filename=$isofile quiet noeject fastboot toram fsck.mode=skip noprompt splash
  initrd (loop)/casper/initrd.lz
}
```
I've made sure the "toram" kernel option is present above. Of course, you will need to update the ISO file path, and ensure the (hd0,5) points to a device/partition number pair that applies to you. After running `sudo update-grub` and rebooting, select Rescuezilla from your GRUB menu.

Once booted, run eg, `mount -l | grep sd` to see that the internal hard drive containing the ISO image is actually mounted: 
```
/dev/sda5 on /isodevice type ext4 (rw,relatime)
```

Given it was booted using the "toram" kernel option, you can safely unmount this device with:
```
# You will need to run this every single time Rescuezilla is booted using this approach
sudo umount -lrf /isodevice
```
Once this is done, you can use Rescuezilla as normal.

Thanks for the question, I didn't know about GRUB's ability to boot from ISO images until now!

**Comment by candrapersada** [comment]
other than set isofile= `"/home/USERNAME/Downloads/rescuezilla-2.1.3-64bit.groovy.iso"`
can or not, set isofile to /dev/sdx other, for example /dev/sda5 ntfs?

**Comment by shasheene** [comment]
@candrapersada, the partition /dev/sda6 should be correspond to (hd0,6), and you can set the isofile to an absolute path (with the forward slash prefix). For example:

set isofile="/rescuezilla-2.1.3-64bit.groovy.iso"
loopback loop **(hd0,6)**$isofile

Also remember to only ever boot it using the "Load into RAM" menu, and always run `sudo umount -lrf /isodevice` as described above.

**Comment by candrapersada** [comment]
whether the information will be stored on the web page or not so that it is easier to remember how to use it?

**Comment by shasheene** [comment]
OK, I will publish the instructions described above on Rescuezilla's GitHub Wiki, and link it from the [frequently asked questions section of the website](https://rescuezilla.com/help.html). I will post an update when this has been completed.

**Comment by candrapersada** [comment]
can you use /dev/sdb or /dev/sdc ntfs for booting from an external hard drive?

**Comment by shasheene** [comment]
Yes, so if the Rescuezilla ISO is inside an NTFS partition that is /dev/sdb3 then you put it as (hd1,3). If it was on /dev/sdc5 you put (hd2,5), and so on. As an example, if the partition you want to put the Rescuezilla ISO on to is "/dev/sdf8" and if you have placed the ISO image under "/home/candapersada/rescuezilla-2.1.3-64bit.groovy.iso" then the following should work:
```
set isofile="/home/candapersada/rescuezilla-2.1.3-64bit.groovy.iso"
loopback loop (hd5,8)$isofile
```

An NTFS filesystem should work, but I do find sometimes Windows locks the NTFS filesystem unless you restart the system into Rescuezilla, because Windows likes to hibernate etc.

**Comment by shasheene** [comment]
> I will publish the instructions described above on Rescuezilla's GitHub Wiki

I have now created a Rescuezilla GitHub Wiki page capturing what has been discussed on this issue: https://github.com/rescuezilla/rescuezilla/wiki/Installing-Rescuezilla-as-a-rescue-partition 

They'll be improved support to Resceuzilla for this behavior in future versions, which will simply some of the instructions over time. For now, I am closing this issue as the question has been answered
